src/taipy/core/_repository/_v2/_abstract_repository.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import json
+import pathlib
+import re
+from abc import abstractmethod
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Any, Dict, Generic, Iterable, List, Optional, TypeVar, Union
+
+Json = Union[dict, list, str, int, float, bool, None]
+
+ModelType = TypeVar("ModelType")
+Entity = TypeVar("Entity")
```

Comment:
@jrobinAV @joaoandre-avaiga 
I noticed we have the use of TypeVar for ModelType and EntityType in many places (_filesystem_repository.py, _manager.py, _sql_repository.py). Does it make sense to define and share them all in 1 place (alias.py perhaps)?

Reply:
Yeah I've moved them to `taipy.core.common.typing`, just forgot to import it on this file. Thanks!

Reply:
I had to rollback this change because mypy is no reconizing TypeVar defined in another file as a generic type, so this breaks the linter.